crates/red_knot_python_semantic/resources/mdtest/snapshots/attribute_assignment.md_-_Attribute_assignment_-_Data_descriptors_-_Invalid_`__set__`_method_signature.snap

```diff
@@ -0,0 +1,39 @@
+---
+source: crates/red_knot_test/src/lib.rs
+expression: snapshot
+---
+---
+mdtest name: attribute_assignment.md - Attribute assignment - Data descriptors - Invalid `__set__` method signature
+mdtest path: crates/red_knot_python_semantic/resources/mdtest/diagnostics/attribute_assignment.md
+---
+
+# Python source files
+
+## mdtest_snippet.py
+
+```
+ 1 | class WrongDescriptor:
+ 2 |     def __set__(self, instance: object, value: int, extra: int) -> None:
+ 3 |         pass
+ 4 | 
+ 5 | class C:
+ 6 |     attr: WrongDescriptor = WrongDescriptor()
+ 7 | 
+ 8 | instance = C()
+ 9 | 
+10 | # TODO: ideally, we would mention why this is an invalid assignment (wrong number of arguments for `__set__`)
+11 | instance.attr = 1  # error: [invalid-assignment]
+```
+
+# Diagnostics
+
+```
+error: lint:invalid-assignment
+  --> /src/mdtest_snippet.py:11:1
+   |
+10 | # TODO: ideally, we would mention why this is an invalid assignment (wrong number of arguments for `__set__`)
+11 | instance.attr = 1  # error: [invalid-assignment]
+   | ^^^^^^^^^^^^^ Invalid assignment to data descriptor attribute `attr` on type `C` with custom `__set__` method
```

Comment:
Not for fixing in this PR, but just noting it seems like we are currently inconsistent about whether we blame an invalid-assignment error on the target, on the RHS, or on the entire expression. I'm seeing that in case of `x: int = "foo"` we mark the entire expression, in `x = "foo"` (with a separate prior `x: int`) we mark just the target, and now here in these attribute cases we also mark just the target. Pyright seems to be consistent about always marking the RHS, and I can see the justification for that.

Reply:
I also thought it would be more intuitive to highlight the RHS, but didn't want to change behavior in my last PR or here. But that should be easy to fix.